docs/Theming.md

```diff
@@ -398,73 +398,24 @@ const ThemeToggler = () => {
 
 ## Conditional Formatting
 
-Sometimes you want the format to depend on the value. The following example shows how to create a new custom `NumberField` component which highlight its text in red when its value is 100 or higher.
+Sometimes you want the format to depend on the value. Use `useRecordContext` to grab the record in a component, and the `sx` prop to apply the format.
 
-{% raw %}
-```jsx
-import * as React from 'react';
-import { NumberField, List, Datagrid, TextField, EditButton } from 'react-admin';
-
-const ColoredNumberFieldStyles = {
-    small: { color: 'black' },
-    big: { color: 'red' },
-};
-
-const ColoredNumberField = (props) => (
-    <NumberField
-        sx={{
-            ...(props.record[props.source] < 100 &&
-                ColoredNumberFieldStyles.small),
-            ...(props.record[props.source] >= 100 &&
-                ColoredNumberFieldStyles.big),
-        }}
-        {...props}
-    />
-);
-
-// Ensure the original component defaultProps are still applied as they may be used by its parents (such as the `Show` component):
-ColoredNumberField.defaultProps = NumberField.defaultProps;
-
-export const PostList = () => (
-    <List>
-        <Datagrid>
-            <TextField source="id" />
-            ...
-            <ColoredNumberField source="nb_views" />
-            <EditButton />
-        </Datagrid>
-    </List>
-);
-```
-{% endraw %}
-
-Furthermore, you may extract this highlighting strategy into a Higher Order Component if you'd like to reuse it for other components as well:
+The following example shows how to create a new `<ColoredNumberField>` component, which renders with red text when its value is less than 0.
 
 {% raw %}
 ```jsx
-import * as React from 'react';
-import { NumberField, List, Datagrid, TextField, EditButton } from 'react-admin';
+import { iseRecordContext, NumberField, List, Datagrid, TextField, EditButton } from 'react-admin';
```

Comment:
@fzaninotto , Actually looks like a typo on `iseRecordContext`

Reply:
Good catch, can you open a PR?

Reply:
https://github.com/marmelab/react-admin/pull/7802